docs/_tooling/extensions/score_metamodel/tests/test_rules_file_based.py

```diff
@@ -0,0 +1,137 @@
+# *******************************************************************************
+# Copyright (c) 2025 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+
+from collections.abc import Callable
+from dataclasses import dataclass, field
+import os
+from sphinx.testing.util import SphinxTestApp
+from sphinx_needs.data import SphinxNeedsData
+import pytest
+from pathlib import Path
+import shutil
+
+RST_DIR = Path(__file__).absolute().parent / "rst"
+DOCS_DIR = Path(__file__).absolute().parent.parent.parent.parent.parent
+TOOLING_DIR_NAME = "_tooling"
+
+RST_FILES = [str(f.relative_to(RST_DIR)) for f in Path(RST_DIR).rglob("*.rst")]
+
+
+@pytest.fixture
+def sphinx_base_dir(tmp_path_factory: pytest.TempPathFactory) -> Path:
+    base_dir: Path = tmp_path_factory.mktemp("docs")
+    shutil.copy(DOCS_DIR / "conf.py", base_dir)
+    shutil.copytree(
+        DOCS_DIR / TOOLING_DIR_NAME,
+        base_dir / TOOLING_DIR_NAME,
+        dirs_exist_ok=True,
+        ignore=shutil.ignore_patterns("*.rst"),
+    )
+    return base_dir
+
+
+@pytest.fixture
+def index_file() -> Callable[[Path], str]:
+    def _create_rst_file(rst_file: Path) -> str:
+        index_rst: str = f"""
+.. toctree::
+   {rst_file.stem}
+"""
+        return index_rst
+
+    return _create_rst_file
+
+
+@pytest.fixture
+def sphinx_app_setup(
+    sphinx_base_dir: Path, index_file: Callable[[Path], str]
+) -> Callable[[Path], SphinxTestApp]:
+    def _create_app(rst_file: Path) -> SphinxTestApp:
+        shutil.copy(rst_file, sphinx_base_dir)
+        index_context: str = index_file(rst_file)
+        (sphinx_base_dir / "index.rst").write_text(index_context)
+        app: SphinxTestApp = SphinxTestApp(
+            freshenv=True,
+            srcdir=sphinx_base_dir,
+            outdir=sphinx_base_dir / "out",
+            buildername="html",
+        )
+        return app
+
+    return _create_app
+
+
+@dataclass
+class InfoElement:
```

Comment:
What is an `InfoElement`? Looks like a wrapper around a need element on a specific line?! -> NeedLocationWithExpectations? Or something better... 😆